dockerworker/dockerimage.go

```diff
@@ -0,0 +1,148 @@
+// +build docker
```

Comment:
The management of docker images should really be a task feature, just like the other features we have (os groups, livelog, taskcluster proxy, mounting files/folders/caches, superseding, chain of trust, interactive task, run as administrator, ...). Each feature is called at task start, task stop, and has a chance to do custom initialisation when the worker starts up. This also means task setup steps happen before the max run timer starts, and failures in initialisation/termination can be propagated back to calling code so that appropriate cleanup happens, task resolution is handled correctly, etc.

This is really no different, it is just a mandatory feature if docker engine is specified.